src/app/redux/UserReducer.js

```diff
@@ -0,0 +1,345 @@
+import {fromJS} from 'immutable';
+import { DEFAULT_LANGUAGE } from 'app/client_config';
+import store from 'store';
+
+// Action constants
+const SHOW_LOGIN = 'user/SHOW_LOGIN';
+const SHOW_TERMS = 'user/SHOW_TERMS';
+const HIDE_LOGIN = 'user/HIDE_LOGIN';
+export const SAVE_LOGIN_CONFIRM = 'user/SAVE_LOGIN_CONFIRM';
+export const SAVE_LOGIN = 'user/SAVE_LOGIN';
+const REMOVE_HIGH_SECURITY_KEYS = 'user/REMOVE_HIGH_SECURITY_KEYS';
+const CHANGE_LANGUAGE = 'user/CHANGE_LANGUAGE';
+const SHOW_TRANSFER = 'user/SHOW_TRANSFER';
+const HIDE_TRANSFER = 'user/HIDE_TRANSFER';
+const SHOW_POWERDOWN = 'user/SHOW_POWERDOWN';
+const HIDE_POWERDOWN = 'user/HIDE_POWERDOWN';
+const SHOW_PROMOTE_POST = 'user/SHOW_PROMOTE_POST';
+const HIDE_PROMOTE_POST = 'user/HIDE_PROMOTE_POST';
+const SET_TRANSFER_DEFAULTS = 'user/SET_TRANSFER_DEFAULTS';
+const CLEAR_TRANSFER_DEFAULTS = 'user/CLEAR_TRANSFER_DEFAULTS';
+const SET_POWERDOWN_DEFAULTS = 'user/SET_POWERDOWN_DEFAULTS';
+const CLEAR_POWERDOWN_DEFAULTS = 'user/CLEAR_POWERDOWN_DEFAULTS';
+export const USERNAME_PASSWORD_LOGIN = 'user/USERNAME_PASSWORD_LOGIN';
+export const SET_USER = 'user/SET_USER';
+const CLOSE_LOGIN = 'user/CLOSE_LOGIN';
+export const LOGIN_ERROR = 'user/LOGIN_ERROR';
+export const LOGOUT = 'user/LOGOUT';
+const SHOW_SIGN_UP = 'user/SHOW_SIGN_UP';
+const HIDE_SIGN_UP = 'user/HIDE_SIGN_UP';
+const KEYS_ERROR = 'user/KEYS_ERROR';
+const ACCOUNT_AUTH_LOOKUP = 'user/ACCOUNT_AUTH_LOOKUP';
+const SET_AUTHORITY = 'user/SET_AUTHORITY';
+const HIDE_CONNECTION_ERROR_MODAL = 'user/HIDE_CONNECTION_ERROR_MODAL';
+const SET = 'user/SET';
+// Saga-related
+export const LOAD_SAVINGS_WITHDRAW = 'user/LOAD_SAVINGS_WITHDRAW';
+export const UPLOAD_IMAGE = 'user/UPLOAD_IMAGE';
+
+const defaultState = fromJS({
+    current: null,
+    show_login_modal: false,
+    show_transfer_modal: false,
+    show_promote_post_modal: false,
+    show_signup_modal: false,
+    pub_keys_used: null,
+    locale: DEFAULT_LANGUAGE
+});
+
+if (process.env.BROWSER) {
+    const locale = store.get('language');
+    if (locale) defaultState.locale = locale;
+}
+
+export default function reducer(state = defaultState, action) {
+    const payload = action.payload;
+
+    switch (action.type) {
```

Comment:
Alphabetize cases.

Reply:
i really don't think alphabetical order makes sense actually. for example, CONFIRM_OPERATION and HIDE_CONFIRM should be adjacent.

Reply:
That suggests inconsistent naming choices. Alphabetization is 'a' stricture that helps us uncover poor choices like that.

Reply:
I'm not going to rename all of the action types in this branch. if you think this is important you can make an issue and the team can discuss.

Reply:
We'll fold that, and the reordering into #2072